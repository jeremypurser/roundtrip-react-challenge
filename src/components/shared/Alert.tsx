import * as React from 'react';
import Toast from 'react-bootstrap/Toast';

export type AlertProps = {
  type: string | undefined;
  message: string | undefined;
  show: boolean;
  onClose: () => void | undefined;
};

export const Alert = (props: AlertProps) => {
  return (
    <Toast
      show={props.show}
      onClose={props.onClose}
      delay={3000}
      autohide={true}
    >
      <Toast.Header>
        <strong className="mr-auto">{props.type}</strong>
      </Toast.Header>
      <Toast.Body>{props.message}</Toast.Body>
    </Toast>
  );
};
