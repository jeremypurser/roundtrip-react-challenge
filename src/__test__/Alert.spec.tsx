import React from 'react';
import renderer from 'react-test-renderer';
import { Alert } from '../components/shared/Alert';

describe('Alert', () => {
  it('should render correctly given valid props', () => {
    const tree = renderer
      .create(
        <Alert
          type="Success"
          message="Request completed"
          show={true}
          onClose={() => {}}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
