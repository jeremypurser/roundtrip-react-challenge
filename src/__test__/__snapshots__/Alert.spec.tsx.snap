// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert should render correctly given valid props 1`] = `
<div
  aria-atomic="true"
  aria-live="assertive"
  className="fade toast show"
  role="alert"
>
  <div
    className="toast-header"
  >
    <strong
      className="mr-auto"
    >
      Success
    </strong>
    <button
      className="close ml-2 mb-1"
      data-dismiss="toast"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
      <span
        className="sr-only"
      >
        Close
      </span>
    </button>
  </div>
  <div
    className="toast-body"
  >
    Request completed
  </div>
</div>
`;
